src/Skia/Avalonia.Skia/Gpu/OpenGl/GlSkiaExternalObjectsFeature.cs

```diff
@@ -240,4 +247,31 @@ public IBitmapImpl SnapshotWithAutomaticSync()
         using (_gpu.EnsureCurrent())
             return TakeSnapshot();
     }
+
+    private int CopyToNewTexture(int textureType, int sourceTextureId, int internalFormat, int width, int height)
+    {
+        var gl = _gpu.GlContext.GlInterface;
+
+        using var _ = _gpu.EnsureCurrent();
+
+        if (_fbo == 0)
+            _fbo = gl.GenFramebuffer();
+
+        // Bind source texture
+        gl.BindFramebuffer(GL_FRAMEBUFFER, _fbo);
+        gl.FramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, textureType, sourceTextureId, 0);
+
+        // Create destination texture
+        var destTextureId = gl.GenTexture();
+        gl.BindTexture(textureType, destTextureId);
+        gl.TexImage2D(textureType, 0, internalFormat, width, height, 0, GL_RGBA, GL_UNSIGNED_BYTE, IntPtr.Zero);
+
+        // Copy
+        gl.BindFramebuffer(GL_READ_FRAMEBUFFER, _fbo);
```

Comment:
Note that `GL_READ_FRAMEBUFFER` is not Available with OpenGLES 2.0. Using `GL_FRAMEBUFFER` is safe since it affects both read and write framebuffers.